### PR TITLE
Removed deprecated API uses in perf_suite, apps, spec-example, perfor…

### PR DESCRIPTION
**Pull request by wrrobin**
…mance dirs

Related to #1025 

Signed-off-by: Md Rahman <md.rahman@intel.com>

### CODE REVIEW

File: test/performance/shmem_perf_suite/latency_common.h [hunk 1]
```diff
@@ -69,12 +66,10 @@ void calc_and_print_results(double start, double end, int len,
         printf("Individual latency for PE %6d is %10.2f\n",
                 metric_info->my_node, latency);
     }
-    shmem_barrier(start_pe, stride, nPEs, bar_psync);
+    shmem_team_sync(streaming_team);
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/latency_common.h]: No quiet needed.

File: test/performance/shmem_perf_suite/bw_common.h [hunk 2]
```diff
@@ -141,18 +141,15 @@ void print_data_results(double bw, double mr, const perf_metrics_t * const data,
...
     static double pe_bw_sum, bw = 0.0; /*must be symmetric for reduction*/
     double pe_bw_avg = 0.0, pe_mr_avg = 0.0;
     int nred_elements = 1;
-    static double pwrk[SHMEM_REDUCE_MIN_WRKDATA_SIZE];
     static double pe_time_start, pe_time_end,
                   end_time_max = 0.0, start_time_min = 0.0;
     double total_t = 0.0, total_t_max = 0.0;
     int multiplier = 1;
 
-    PE_set_used_adjustments(&nPEs, &stride, &start_pe, metric_info);
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/bw_common.h:154]: I have reverted some of the changes. Please review when you can.

File: test/performance/shmem_perf_suite/bw_common.h [hunk 3]
```diff
@@ -188,16 +185,12 @@ void calc_and_print_results(double end_t, double start_t, int len,
...
         if (nPEs >= 2) {
-            shmem_double_min_to_all(&start_time_min, &pe_time_start, nred_elements,
-                                start_pe, stride, nPEs, pwrk,
-                                red_psync);
-            shmem_barrier(start_pe, stride, nPEs, bar_psync);
-            shmem_double_max_to_all(&end_time_max, &pe_time_end, nred_elements,
-                                start_pe, stride, nPEs, pwrk,
-                                red_psync);
+            shmem_double_min_reduce(streaming_team, &start_time_min, &pe_time_start, nred_elements);
+            shmem_team_sync(streaming_team);
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/bw_common.h]: I was wrong, no quiet needed.

File: test/performance/shmem_perf_suite/common.h [hunk 4]
```diff
@@ -781,24 +769,6 @@ red_PE_set validation_set(perf_metrics_t * const my_info, int *nPEs)
     }
 }
 
-/* reduction to collect performance results from PE set
- * then start_pe will print results --- assumes num_pes is even */
-static inline
-void PE_set_used_adjustments(int *nPEs, int *stride, int *start_pe,
-                             perf_metrics_t * const my_info) {
-    red_PE_set PE_set = validation_set(my_info, nPEs);
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/common.h:789]: Done.

File: test/performance/shmem_perf_suite/common.h [hunk 5]
```diff
@@ -664,7 +652,7 @@ int check_hostname_validation(const perf_metrics_t * const my_info) {
     shmem_barrier_all();
 
     /* nelems needs to be updated based on 32-bit API */
-    shmem_char_fcollect(SHMEM_TEAM_WORLD, dest, hostname, hostname_size/4);
+    shmem_char_fcollect(SHMEM_TEAM_WORLD, dest, hostname, hostname_size);
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/common.h:655]: Oh.. yes.. I missed that.. Will check and remove.

File: test/performance/shmem_perf_suite/common.h [hunk 6]
```diff
@@ -781,11 +769,8 @@ red_PE_set validation_set(perf_metrics_t * const my_info, int *nPEs)
     }
 }
 
-/* reduction to collect performance results from PE set
- * then start_pe will print results --- assumes num_pes is even */
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/common.h:785]: Yes, odd PEs will never come up to this point.

File: test/performance/shmem_perf_suite/common.h [hunk 7]
```diff
@@ -815,3 +798,40 @@ void print_header(perf_metrics_t * const metric_info) {
...
+
+    int my_pe = metric_info->my_node;
+    if (target_team == SHMEM_TEAM_INVALID && (my_pe >= metric_info->midpt && my_pe < metric_info->num_pes)) {
+        fprintf(stderr, "PE %d: Target team creation failed\n", metric_info->my_node);
+        return -1;
+    }
+
+    return 0;
+}
+
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/common.h:829]: Yes, `create_teams` will not be executed if num_pes is odd. So, I think its best to remove that comment before. And, I will add that comment in the place where we are actually checking it.

File: test/performance/tests/msgrate.c [hunk 8]
```diff
@@ -108,23 +104,18 @@ static void
...
-        pe_size = world_size - 1;
-    }
-
     if (!(world_size % 2 == 1 && rank == (world_size - 1))) {
         if (rank < world_size / 2) {
             for (i = 0 ; i < niters ; ++i) {
                 cache_invalidate();
 
-                shmem_barrier(0, 0, pe_size, barrier_pSync);
+                shmem_barrier_all();
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/tests/msgrate.c:118]: Oh.. I never checked this one.. this is out of perf suite.. and I am not sure why we would need a separate test when we report message rate in the perf suite tests. Anyway, I will check. 

We can add one row.. may be I can add that in this PR itself.

File: test/performance/shmem_perf_suite/common.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/common.h:655]: Done

File: test/performance/shmem_perf_suite/common.h (hunk 7 above)
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/shmem_perf_suite/common.h:829]: Added.

File: test/performance/tests/msgrate.c (hunk 8 above)
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/tests/msgrate.c:118]: Yes.. good catch about the deadlock. Will correct.

[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/tests/msgrate.c]: @davidozog I have updated `msgrate` test to make sure it runs with the default setting. This way, we can still keep it and it produces valid numbers. Please review and let me know if it looks ok.

File: test/performance/tests/msgrate.c [hunk 9]
```diff
@@ -108,23 +104,26 @@ static void
...
 
     shmem_barrier_all();
 
-    if (world_size % 2 == 1) {
-        pe_size = world_size - 1;
-    }
+    shmem_team_t sync_team;
+    shmem_team_config_t *config = NULL;
+    if (world_size % 2 == 1)
+        shmem_team_split_strided(SHMEM_TEAM_WORLD, 0, 1, world_size - 1, config, 0, &sync_team);
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/tests/msgrate.c]: Done.

File: test/performance/tests/msgrate.c [hunk 10]
```diff
@@ -267,35 +267,17 @@ main(int argc, char *argv[])
...
+                fprintf(stderr, "Error: job size (%d) < number of peers (%d)\n",
+                                 world_size, npeers);
                 start_err = 77;
             } else if (ppn < 1) {
                 fprintf(stderr, "Error: must specify process per node (-n #)\n");
                 start_err = 77;
-            } else if (world_size / ppn <= npeers) {
-                fprintf(stderr, "Error: node count <= number of peers\n");
-                start_err = 77;
             }
```
[the remark this answers is not in the document]
Comment (wrrobin) [reply test/performance/tests/msgrate.c:279]: Agree. Please check the last change.

**Comment by davidozog** [comment]
LGTM!  Weird that `shmemlatency` takes so long though...?

**Review by davidozog — APPROVED** [review]
Thanks @wrrobin

**Comment by wrrobin** [comment]
@kholland-intel Please let me know if this looks ok to you as well.

**Review by kholland-intel — APPROVED** [review]
LGTM, sorry for the delay.

**Pull request merged**